src/bridge/client/client.rs

```diff
@@ -161,18 +164,29 @@ impl BitVMClient {
         }
     }
 
-    // fn verify_data(&self, data: &BitVMClientData) {
-    //     for peg_in_graph in data.peg_in_graphs.iter() {
-    //         self.verify_peg_in_graph(peg_in_graph);
-    //     }
-    //     for peg_out_graph in data.peg_out_graphs.iter() {
-    //         self.verify_peg_out_graph(peg_out_graph);
-    //     }
-    // }
-
-    // fn verify_peg_in_graph(&self, peg_in_graph: &PegInGraph) {}
+    fn validate_data(&self, data: &BitVMClientData) -> bool {
+        for peg_in_graph in data.peg_in_graphs.iter() {
+            if !peg_in_graph.validate() {
+                println!(
+                    "Encountered invalid peg in graph (Graph id: {})",
+                    peg_in_graph.id()
+                );
+                return false;
```

Comment:
[Minor] Consider just flagging an invalid graph and continue validation. Just thinking about a scenario with many graphs where just one graph that was executed in the past is corrupted. Do we want a hard fail in this case? Perhaps fail only on graphs in progress?